src/site/mdbook.rs

```diff
@@ -0,0 +1,145 @@
+use axoasset::{LocalAsset, SourceFile};
+use camino::{Utf8Path, Utf8PathBuf};
+use mdbook::MDBook;
+
+use crate::config::MdBookConfig;
+use crate::errors::*;
+use crate::message::{Message, MessageType};
+use crate::site::Site;
+
+use toml_edit::value;
+
+use super::markdown::SyntaxTheme;
+
+/// Name of the config for mdbook
+const MDBOOK_TOML: &str = "book.toml";
+
+// Files we're importing
+const THEME_GENERAL_CSS_PATH: &str = "css/general.css";
+const THEME_GENERAL_CSS: &str = include_str!("../../oranda-css/mdbook-theme/css/general.css");
+const THEME_VARIABLES_CSS_PATH: &str = "css/variables.css";
+const THEME_VARIABLES_CSS: &str = include_str!("../../oranda-css/mdbook-theme/css/variables.css");
+const THEME_CHROME_CSS_PATH: &str = "css/chrome.css";
+const THEME_CHROME_CSS: &str = include_str!("../../oranda-css/mdbook-theme/css/chrome.css");
+const THEME_FONTS_CSS_PATH: &str = "fonts/fonts.css";
+const THEME_FONTS_CSS: &str = include_str!("../../oranda-css/mdbook-theme/fonts/fonts.css");
+const THEME_BOOK_JS_PATH: &str = "book.js";
+const THEME_BOOK_JS: &str = include_str!("../../oranda-css/mdbook-theme/book.js");
+const THEME_INDEX_HBS_PATH: &str = "index.hbs";
+const THEME_INDEX_HBS: &str = include_str!("../../oranda-css/mdbook-theme/index.hbs");
+
+const THEME_HIGHLIGHT_CSS_PATH: &str = "highlight.css";
+const THEMES: &[(&str, &str)] = &[(
+    "MaterialTheme",
+    include_str!("../../oranda-css/mdbook-theme/highlight-js-themes/base16-material.css"),
+)];
+
+/// Build and write mdbook to the dist dir
+pub fn build_mdbook(
+    dist: &Utf8Path,
+    book_cfg: &MdBookConfig,
+    syntax_theme: &SyntaxTheme,
+) -> Result<()> {
+    Message::new(MessageType::Info, "Building mdbook...").print();
+    tracing::info!("Building mdbook...");
+
+    // Read mdbook's config to find the right dirs
+    let book_dir = Utf8PathBuf::from(&book_cfg.path);
+
+    if book_cfg.theme.unwrap_or(true) {
```

Comment:
Can this be `Some(false)` or does this condition always pass? As in, is there a chance someone's gonna put in `theme: false` in their config?

Reply:
yes, theme = false is basically the only value we expect people to manually set (or in the future, a theme name, maybe).